app/scripts/modules/core/src/pipeline/config/pipelineConfigurer.js

```diff
@@ -365,6 +368,19 @@ module.exports = angular.module('spinnaker.core.pipeline.config.pipelineConfigur
       return msg;
     };
 
+    this.getPipelineExecutions = () => {
+      executionService.getExecutionsForConfigIds($scope.pipeline.application, $scope.pipeline.id, 5)
+        .then(executions => {
+          $scope.pipelineExecutions = executions;
+        })
+        .catch(() => $scope.pipelineExecutions = []);
+      return $scope.pipelineExecutions;
+    };
+
+    $scope.changePipelineExecution = executionId => {
+      $location.search('executionId', executionId);
```

Comment:
This should either be a `$state.go` or if possible you should be able to include a `ui-sref` in your template that updates the query parameter.